src/Shared/CommunicationsUtilities.cs

```diff
@@ -104,7 +123,7 @@ public override string ToString()
             return String.Format("{0} {1} {2} {3} {4} {5} {6}", options, salt, fileVersionMajor, fileVersionMinor, fileVersionBuild, fileVersionPrivate, sessionId);
         }
 
-        internal int[] RetrieveHandshakeComponents()
+        public int[] RetrieveHandshakeComponents()
```

Comment:
Since Handshake and IHandshake is internal, does this need to be public?

Reply:
The interface declares this method, so yes, it must be public.